src/main/java/org/opensearch/eval/engine/OpenSearchEngine.java

```diff
@@ -218,6 +227,94 @@ public Double getJudgmentValue(final String judgmentsId, final String userQuery,
 
     }
 
+    @Override
+    public Map<String, Long> getRandomUbiQueries(final int n) throws IOException {
+
+        final long seed = System.currentTimeMillis();
+        final RandomScoreFunction randomScoreFunction = new RandomScoreFunction.Builder().seed(String.valueOf(seed)).field(USER_QUERY_FIELD).build();
+        final FunctionScore functionScore = new FunctionScore.Builder().randomScore(randomScoreFunction).build();
+
+        final MatchAllQuery matchAllQuery = new MatchAllQuery.Builder().build();
+
+        final FunctionScoreQuery functionScoreQuery = new FunctionScoreQuery.Builder()
+                .query(matchAllQuery.toQuery())
+                .functions(List.of(functionScore))
+                .build();
+
+        final ExistsQuery existsQuery = new ExistsQuery.Builder()
+                .field(USER_QUERY_FIELD)
+                .build();
+
+        final BoolQuery boolQuery = new BoolQuery.Builder()
+                .must(q -> q.exists(existsQuery))
+                .must(q -> q.functionScore(functionScoreQuery))
+                .build();
+
+        final SearchRequest searchRequest = new SearchRequest.Builder()
+                .index(Constants.UBI_QUERIES_INDEX_NAME)
+                .query(boolQuery.toQuery())
+                .collapse(FieldCollapse.of(c -> c.field(USER_QUERY_FIELD)))
+                .size(n)
+                .build();
+
+        final SearchResponse<UbiQuery> searchResponse = client.search(searchRequest, UbiQuery.class);
+
+        final Map<String, Long> querySet = new HashMap<>();
+
+        // Is having the frequency for the random queries useful?
```

Comment:
I would index the frequency to be consistent with the other sampling methods if there are no strong reasons not to do so. It definitely is less important but might still be useful/interesting for users.

Reply:
`random` sampler is not indexing the frequency, unresolving this comment